SHOWRAW.BAS

```diff
@@ -1,32 +1,116 @@
-REM A script to show the raw screen data saved in MANDELBR.BAS
-
 DECLARE SUB LoadRawScreen (filename AS STRING, w AS INTEGER, h AS INTEGER)
-
+DECLARE SUB CyclePalette ()
 SCREEN 13
 CLS
-LINE(0,0)-(320-200), 15, BF
+LINE (0, 0)-(320, 200), 15, BF
+CALL LoadRawScreen("C:\SCREEN.RAW", 320, 200)
 
-CALL LoadRawScreen("SCREEN.RAW", 320, 200)
+CALL CyclePalette
 
-SUB LoadRawScreen(filename AS STRING, w AS INTEGER, h AS INTEGER)
+SUB CyclePalette
+        DIM blue AS LONG
+        DIM green AS LONG
+        DIM red AS LONG
+        DIM pcolour AS LONG
+        DIM pvalues1(255) AS LONG
+        DIM pvalues2(255) AS LONG
+        DIM pvalues3(255) AS LONG
+        DIM pvalues4(255) AS LONG
+        FOR i = 0 TO 63
+                blue = i
+                green = i / 2
+                red = i / 3
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues1(i) = pcolour
+                pvalues1(i + 128) = pcolour
+                blue = i / 2
+                green = i / 3
+                red = i
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues2(i) = pcolour
+                pvalues2(i + 128) = pcolour
+                blue = i / 3
+                green = i
+                red = i / 2
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues3(i) = pcolour
+                pvalues3(i + 128) = pcolour
+                blue = i
+                green = i
+                red = i
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues4(i) = pcolour
+                pvalues4(i + 128) = pcolour
+        NEXT i
+        FOR i = 63 TO 0 STEP -1
+                blue = i
+                green = i / 2
+                red = i / 3
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues1(i + 64) = pcolour
+                pvalues1(i + 192) = pcolour
+                blue = i / 2
+                green = i / 3
+                red = i
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues2(i + 64) = pcolour
+                pvalues2(i + 192) = pcolour
+                blue = i / 3
+                green = i
+                red = i / 2
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues3(i + 64) = pcolour
+                pvalues3(i + 192) = pcolour
+                blue = i
+                green = i
+                red = i
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues4(i + 64) = pcolour
+                pvalues4(i + 192) = pcolour
```

Comment:
Looks like the only difference between this block and the previous is the pvaluesN(i + offset).
Where offset is 0 originally but now 64 here.
If this is annoying to refactor I wouldn't bother, duplication isn't a big deal.

Reply:
Nope... A closer look reveals the second one steps backward. It's creating a rising and then falling colour gradient to fill up the 256 range twice with two peaks.